main/src/ca/uwaterloo/flix/language/phase/Desugar.scala

```diff
@@ -1135,7 +1135,7 @@ object Desugar {
     // ignore redundant effect casts to IO
     val correctedEff = eff match {
       case None => None
-      case Some(DesugaredAst.Type.Ambiguous(Name.QName(_, Name.NName(Nil, _), Name.Ident(_, "IO", _), _), _)) => None
+      case Some(DesugaredAst.Type.Ambiguous(Name.QName(Name.NName(Nil, _), Name.Ident(_, "IO", _), _), _)) => None
```

Comment:
Can we just check that the name is unqualified and matches IO? This is hard to read.

Reply:
I suppose I can refactor.